src/sentry/api/serializers/rest_framework/notification_action.py

```diff
@@ -43,16 +43,48 @@ class NotificationActionSerializer(CamelSnakeModelSerializer):
     Django Rest Framework serializer for incoming NotificationAction API payloads
     """
 
-    integration_id = serializers.IntegerField(required=False)
-    sentry_app_id = serializers.IntegerField(required=False)
-    projects = serializers.ListField(child=ProjectField(scope="project:read"), required=False)
-
-    service_type = serializers.CharField()
-    target_type = serializers.CharField()
-    trigger_type = serializers.CharField()
-
-    target_identifier = serializers.CharField(required=False)
-    target_display = serializers.CharField(required=False)
+    # Optional and not needed for spike protection so not documenting
+    integration_id = serializers.IntegerField(
+        required=False,
+    )
+
+    # Optional and not needed for spike protection so not documenting
+    sentry_app_id = serializers.IntegerField(
+        required=False,
+    )
+    projects = serializers.ListField(
+        help_text="""List of serialized projects that Notificaton Action is created for""",
+        child=ProjectField(scope="project:read"),
+        required=False,
+    )
+
+    service_type = serializers.CharField(
+        help_text="Service that is used for sending the notification\n"
+        + """- `email`\n"""
+        + """- `slack`\n"""
+        + """- `pagerduty`\n"""
+        + """- `opsgenie`\n"""
+        + """- `sentry_notification`\n"""
+    )
+    target_type = serializers.CharField(
+        help_text="""Type of notification recipient\n"""
+        + """ - `specific`\n"""
+        + """ - `user`\n"""
+        + """ - `team`\n"""
+        + """ - `sentry_app`\n"""
+    )
+    trigger_type = serializers.CharField(
+        help_text="""Type of the trigger that causes the notification. The only supported trigger right now is: `spike-protection`"""
+    )
+
+    target_identifier = serializers.CharField(
+        help_text="""ID of the notification target. For example, Slack channel ID""",
+        required=False,
+    )
+    target_display = serializers.CharField(
+        help_text="""Name of the notification target. For example, Slack channel name""",
+        required=False,
+    )
```

Comment:
slack and pagerduty require `target_identifier` and `target_display`. i'm not sure how to best surface that some of these are required for certain services and not required for others

Reply:
This is the way. By over-documenting in the help_text. Are those the only ones?

Reply:
It looks like discord also requires `target_display` and `target_identifier`. I'm seeing this from the `validate_slack_channel`, `validate_pagerduty_service`, and`validate_discord_channel` methods.